vue/src/components/thread/attachment_list.vue

```diff
@@ -1,25 +1,40 @@
 <script lang="coffee">
-
+import EventBus       from '@/shared/services/event_bus'
+import AbilityService from '@/shared/services/ability_service'
 import prettyBytes from 'pretty-bytes'
 
 export default
   props:
     attachments: [Array, Object]
+    group:
+      default: null
+      type: Object
   methods:
     prettifyBytes: (s) -> prettyBytes(s)
+    deleteDocument: ->
+      EventBus.$emit 'openModal',
+        component: 'GroupForm'
+        props:
+          group: @group
+  computed:
+    canDelete: ->
+      @group && AbilityService.canEditGroup(@group)
 </script>
 <template lang="pug">
 .attachment-list
   .attachment-list__item(v-for="attachment in attachments" :key="attachment.id")
     a.lmo-pointer(v-if="attachment.preview_url" :href='attachment.download_url', target='_blank')
       img.attachment-list__preview(:src="attachment.preview_url")
     .attachment-list__item-details
-      v-icon(v-if="attachment.icon") {{ `mdi-${attachment.icon}` }}
+      v-icon.mr-2(v-if="attachment.icon") {{ `mdi-${attachment.icon}` }}
       div
         a(:href="attachment.download_url")
           span {{ attachment.filename }}
         space
         span.lmo-grey-on-white ({{ prettifyBytes(attachment.byte_size) }})
+        v-btn.ml-2(v-if="group && canDelete" icon :aria-label="$t('common.action.delete')" @click='deleteDocument')
```

Comment:
why do we need a group for this? If this could work wherever there are document_lists, and did not need a group modal, that would be better

Reply:
ah yes i know what i was trying to do. will remove, unnecessary